source/features/quick-repo-deletion.tsx

```diff
@@ -1,133 +1,15 @@
-		event.delegateTarget.open = false;
-		return;
-	}
-
-	// Without the timeout, the same toggle event will also trigger the AbortController
-	setTimeout(start, 1, event.delegateTarget);
-}
 
-async function verifyScopesWhileWaiting(abortController: AbortController): Promise<void> {
-	try {
-		await expectTokenScope('delete_repo');
-	} catch (error) {
-		assertError(error);
-		abortController.abort();
-		await addNotice([
-			'Could not delete the repository. ',
-			parseBackticks(error.message),
-		], {
-			type: 'error',
-			action: (
-				<a className="btn btn-sm primary flash-action" href="https://github.com/settings/tokens">
-					Update token…
-				</a>
-			),
-		});
-	}
-}
-
-async function buttonTimeout(buttonContainer: HTMLDetailsElement): Promise<boolean> {
-	const abortController = new AbortController();
-	// Add a global click listener to avoid potential future issues with z-index
-	document.addEventListener('click', event => {
-		event.preventDefault();
-		abortController.abort();
-		buttonContainer.open = false;
-	}, {once: true});
-
-	void verifyScopesWhileWaiting(abortController);
-
-	let secondsLeft = 5;
-	const button = $('.btn', buttonContainer)!;
-	try {
-		do {
-			button.style.transform = `scale(${1.2 - ((secondsLeft - 5) / 3)})`; // Dividend is zoom speed
-			button.textContent = `Deleting repo in ${pluralize(secondsLeft, '$$ second')}. Cancel?`;
-			await delay(1000, {signal: abortController.signal}); // eslint-disable-line no-await-in-loop
-		} while (--secondsLeft);
-	} catch {
-		button.textContent = 'Delete fork';
-		button.style.transform = '';
-	}
-
-	return !abortController.signal.aborted;
-}
-
-async function start(buttonContainer: HTMLDetailsElement): Promise<void> {
-	if (!await buttonTimeout(buttonContainer)) {
-		return;
-	}
-
-	$('.btn', buttonContainer)!.textContent = 'Deleting repo…';
-	const {nameWithOwner, owner} = getRepo()!;
-	try {
-		await api.v3('/repos/' + nameWithOwner, {
-			method: 'DELETE',
-			json: false,
-		});
-	} catch (error) {
-		assertError(error);
-		buttonContainer.closest('li')!.remove(); // Remove button
-		await addNotice([
-			'Could not delete the repository. ',
-			(error as RefinedGitHubAPIError).response?.message ?? error.message,
-		], {
-			type: 'error',
-		});
-
-		throw error;
-	}
-
-	const forkSource = '/' + getForkedRepo()!;
-	const restoreURL = pageDetect.isOrganizationRepo()
-		? `/organizations/${owner}/settings/deleted_repositories`
-		: '/settings/deleted_repositories';
-	const otherForksURL = `/${owner}?tab=repositories&type=fork`;
-	await addNotice(
-		<><TrashIcon /> <span>Repository <strong>{nameWithOwner}</strong> deleted. <a href={restoreURL}>Restore it</a>, <a href={forkSource}>visit the source repo</a>, or see <a href={otherForksURL}>your other forks.</a></span></>,
```

Comment:
I should probably restore this and show it on the regular deletion confirmation page

https://github.com/refined-github/refined-github/pull/5651


![Screen Shot 19](https://user-images.githubusercontent.com/1402241/169678812-4ebe2cff-b7a4-422d-a9f8-69774867f951.png)

Reply:
Extracted to https://github.com/refined-github/refined-github/issues/7839